packages/client/components/DateBox.tsx

```diff
@@ -0,0 +1,67 @@
+import { Pressable, StyleSheet } from "react-native";
+import { Text, Box, HStack } from "native-base";
+import React, { useState } from "react";
+import DateTimePickerModal from "react-native-modal-datetime-picker";
+import { FontAwesomeIcon } from "@fortawesome/react-native-fontawesome";
+import { falCalendar } from "./icons/light/falCalendar";
+
+type DateBoxProps = {
+    label: string;
+    date: Date | null;
+    setDate: React.Dispatch<React.SetStateAction<Date | null>>;
+    maximumDate?: Date | null;
+    minimumDate?: Date | null;
+    style?: any;
+};
+
+export default function DateBox({ label, style, date, setDate, maximumDate, minimumDate }: DateBoxProps) {
+    const [isDatePickerVisible, setDatePickerVisibility] = useState(false);
+
+    const showDatePicker = () => {
+        setDatePickerVisibility(true);
+    };
+
+    const hideDatePicker = () => {
+        setDatePickerVisibility(false);
+    };
+
+    const handleConfirm = (date: Date) => {
+        setDate(date);
+        hideDatePicker();
+    };
+
+    return (
+        <Box style={{ ...styles.box, ...style }}>
+            <Pressable onPress={showDatePicker}>
+                <HStack justifyContent="space-between">
+                    <Text color={date ? "text.600" : "text.300"}>
```

Comment:
I don't believe text.600 is used for the date anywhere.

```suggestion
                    <Text color={date ? "text.700" : "text.300"}>
```